dll/kernel32.cpp

```diff
@@ -781,6 +921,39 @@ namespace kernel32 {
 		return cbMultiByte;
 	}
 
+	unsigned int WIN_FUNC MultiByteToWideChar(unsigned int codePage, unsigned int dwFlags, const char *lpMultiByteStr, int cbMultiByte, uint16_t *lpWideCharStr, int cchWideChar) {
+		DEBUG_LOG("MultiByteToWideChar(codePage=%u, dwFlags=%u, multiByte=%d, wideChar=%d)\n", codePage, dwFlags, cbMultiByte, cchWideChar);
+
+		// assert (dwFlags == 1); // MB_PRECOMPOSED
+		int i = 0;
+		if (lpWideCharStr == 0) { // return required buffer length
+			while (lpMultiByteStr[i] != 0) {
+				i++;
+			}
+		} else { // else copy source into destination
+			while (i < cchWideChar) {
+				lpWideCharStr[i] = lpMultiByteStr[i];
+				i++;
+			}
+			lpWideCharStr[cchWideChar] = 0; // NUL terminate
+		}
+		return i + 1;
+	}
+
+	unsigned int WIN_FUNC GetStringTypeW(unsigned int dwInfoType, const char *lpSrcStr, int cchSrc, uint16_t *lpCharType) {
```

Comment:
This function looks wrong. `lpSrcStr` should be `uint16_t *`, and it shouldn't copy the string but rather classify characters as lowercase, uppercase, ...